lib/scopedstorage.ts

```diff
@@ -0,0 +1,37 @@
+import NextcloudStorage from './storage'
+
+export default class ScopedStorage implements NextcloudStorage {
+
+    public static GLOBAL_SCOPE_VOLATILE = 'nextcloud_vol'
+    public static GLOBAL_SCOPE_PERSISTENT = 'nextcloud_per'
+    private scope: string
+    private wrapped: Storage
+
+    constructor(scope: string, wrapped: Storage, persistent: boolean) {
+        this.scope = `${persistent ? ScopedStorage.GLOBAL_SCOPE_PERSISTENT : ScopedStorage.GLOBAL_SCOPE_VOLATILE}_${scope}`
+        this.wrapped = wrapped
+    }
+
+    private scopeKey(key: string) {
+        return `${this.scope}_${key}`
+    }
+
+    setItem(key: string, value: string): void {
+        this.wrapped.setItem(this.scopeKey(key), value)
+    }
+
+    getItem(key: string): string | null {
+        return this.wrapped.getItem(this.scopeKey(key))
+    }
+
+    removeItem(key: string): void {
+        this.wrapped.removeItem(this.scopeKey(key))
+    }
+
+    clear(): void {
+        Object.keys(this.wrapped)
+            .filter(key => key.startsWith(this.scope))
```

Comment:
```suggestion
            .filter(key => key.startsWith(`${this.scope}_`))
```

To avoid `abc` deleting all keys of `abcd`, etc

But that still seems to cause an issue with e.g. `files` and `files_pdfviewer`
can we seperate the scopes with `/` or `#` or something else not allowed in appids?

Reply:
Very good point. We could also base64 encode or hash the app ID

Reply:
Also fine, base64 and then `_` added here sounds good

Reply:
Fixed